api/manifest/parser/segment-information.js

```diff
@@ -163,6 +163,10 @@ const SegmentInformation = (function () {
     const segmentDuration = parseInt(this.segmentTemplate.attributes.getNamedItem("duration").nodeValue);
     const segmentTimescale = (this.segmentTemplate.attributes.getNamedItem("timescale")) ? parseInt(
         this.segmentTemplate.attributes.getNamedItem("timescale").nodeValue) : 1;
+
+    // use math.ceil, instead of math.floor, because floor returns the largest integer less than or equal to a given number.
+    // math.ceil returns the smallest integer greater than or equal to a given number.
+    // using math.floor may lead to an incorrect number of segments.
```

Comment:
I think tracking previous errors with comments might not be needed, would rather have a unit test for that if you think it is important, do you agree ?

Reply:
The comment was added before you make the change in master, to explain why ceil has to  be called instead of floor.
It can be removed